exotica_core/src/kinematic_tree.cpp

```diff
@@ -995,6 +997,79 @@ void KinematicTree::SetJointLimitsUpper(Eigen::VectorXdRefConst upper_in)
     UpdateJointLimits();
 }
 
+void KinematicTree::SetJointVelocityLimits(Eigen::VectorXdRefConst velocity_in)
+{
+    if (velocity_in.rows() == num_controlled_joints_)
+    {
+        for (unsigned int i = 0; i < num_controlled_joints_; ++i)
+        {
+            controlled_joints_[i].lock()->velocity_limit = velocity_in(i);
+        }
+    }
+    else
+    {
+        ThrowPretty("Got " << velocity_in.rows() << " but " << num_controlled_joints_ << " expected.");
+    }
+
+    UpdateJointLimits();
+}
+
+void KinematicTree::SetJointAccelerationLimits(Eigen::VectorXdRefConst acceleration_in)
+{
+    if (acceleration_in.rows() == num_controlled_joints_)
+    {
+        for (unsigned int i = 0; i < num_controlled_joints_; ++i)
+        {
+            controlled_joints_[i].lock()->acceleration_limit = acceleration_in(i);
+        }
+
+        has_acceleration_limit = true;
+    }
+    else
+    {
+        ThrowPretty("Got " << acceleration_in.rows() << " but " << num_controlled_joints_ << " expected.");
+    }
+
+    UpdateJointLimits();
+}
+
+void KinematicTree::SetFloatingBaseLimitsPosXYZEulerZYX(
+    const std::vector<double>& lower, const std::vector<double>& upper, const std::vector<double>& velocity, const std::vector<double>& acceleration)
+{
+    bool velocity_acceleration_to_inf = false;
+    if (controlled_base_type_ != BaseType::FLOATING)
+    {
+        ThrowPretty("This is not a floating joint!");
+    }
+    if (velocity.size() == 0 && acceleration.size() == 0)
```

Comment:
If you are adding a second overloaded method you dont need this special handling at all. If signature using velocity & acceleration passed in, set it. If not (old signature), ignore and dont update at all

Reply:
Resolved on WhatsApp